<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <!-- Provide a default empty list for bag paths -->
  <arg name="data_containing_rosbag_paths" default="/home/tutuna/Videos/stim_bug/2024-11-18-17-31-36/2024-11-18-17-31-36_boxi_tf_pure_perception.bag /home/tutuna/Videos/stim_bug/2024-11-18-17-31-36/2024-11-18-17-31-36_jetson_zed2i_images.bag"/>

  <param name="/use_sim_time" value="true"/>
  <arg name="enable_viz" default="true" />

  <node name="zed_left_rgb" 
        pkg="image_transport" 
        type="republish" 
        args="compressed in:=/boxi/zed2i/left/image_raw raw out:=/boxi/zed2i/left/image_raw"
        required="true" />

  <node name="zed_right_rgb" 
        pkg="image_transport" 
        type="republish" 
        args="compressed in:=/boxi/zed2i/right/image_raw raw out:=/boxi/zed2i/right/image_raw"
        required="true" />

  <node name="zed_depth" 
        pkg="image_transport" 
        type="republish" 
        args="compressedDepth in:=/boxi/zed2i/depth/image_raw raw out:=/boxi/zed2i/depth/image_raw"
        required="true" />
    
  <node pkg="rviz" 
        name="rviz" 
        type="rviz" 
        args="-d '$(find box_auto)/rviz/rviz_overlay.rviz'" 
        required="true" />
    
  <group if="$(arg enable_viz)">
    <!-- RQT IM, left RGB view -->
    <node name="image_view_left" 
          pkg="image_view" 
          type="image_view" 
          respawn="false" 
          output="screen">
      <param name="image_transport" value="compressed" />
      <remap from="image" to="/boxi/zed2i/left/image_raw" />           
    </node> 

    <!-- RQT IM, right RGB view -->
    <node name="image_view_right"
          pkg="image_view" 
          type="image_view" 
          respawn="false" 
          output="screen">
      <param name="image_transport" value="compressed" />
      <remap from="image" to="/boxi/zed2i/right/image_raw" />           
    </node> 

    <!-- Depth RVL view -->
    <node name="depth_rvl_view" 
          pkg="image_view" 
          type="image_view" 
          respawn="false" 
          output="screen">
      <param name="image_transport" value="raw" />
      <remap from="image" to="/boxi/zed2i/depth/image_raw" />
      <param name="colormap" value="2" />
      <param name="min_image_value" value="0.5" />
      <param name="max_image_value" value="15.0" />
      <param name="dynamic_range" value="false" />
      <param name="autosize" value="true" />
    </node> 
  </group>

  <node pkg="rosbag" 
        type="play" 
        name="player" 
        output="screen" 
        launch-prefix="xterm -bg lightblue -fg darkslategrey -bd plum -bw 10 -geometry 80x40-50+250 -e" 
        args="--clock -r 1 $(arg data_containing_rosbag_paths)" />

</launch>
